Sources/MapboxCoreNavigation/NativeHandlersFactory.swift

```diff
@@ -87,10 +92,13 @@ class NativeHandlersFactory {
             ]
         ]
         
-        var customConfig = ""
-        if let jsonDataConfig = try? JSONSerialization.data(withJSONObject: historyAutorecordingConfig, options: []),
+        var customConfig = UserDefaults.standard.dictionary(forKey: customConfigKey) ?? [:]
+        customConfig.deepMerge(with: historyAutorecordingConfig)
+        
+        var customConfigJSON = ""
+        if let jsonDataConfig = try? JSONSerialization.data(withJSONObject: customConfig, options: []),
```

Comment:
Is this expected that `customConfig` serialisation can fail? Not sure about putting an assert, because they can distract developer flows, but at very least we can put something in logs to help catch that something is wrong.